dbt/include/bigquery/macros/catalog/catalog.sql

```diff
@@ -1,31 +1,31 @@
 {% macro _bigquery__get_table_shards_sql(information_schema) %}
     select
-        tables.project_id as table_catalog,
-        tables.dataset_id as table_schema,
-        coalesce(REGEXP_EXTRACT(tables.table_id, '^(.+)[0-9]{8}$'), tables.table_id) as table_name,
-        tables.table_id as shard_name,
-        REGEXP_EXTRACT(tables.table_id, '^.+([0-9]{8})$') as shard_index,
-        REGEXP_CONTAINS(tables.table_id, '^.+[0-9]{8}$') and tables.type = 1 as is_date_shard,
-        case
-            when materialized_views.table_name is not null then 'materialized view'
-            when tables.type = 1 then 'table'
-            when tables.type = 2 then 'view'
+        tables.table_catalog,
+        tables.table_schema,
+        coalesce(REGEXP_EXTRACT(tables.table_name, '^(.+)[0-9]{8}$'), tables.table_name) as table_name,
+        tables.table_name as shard_name,
+        REGEXP_EXTRACT(tables.table_name, '^.+([0-9]{8})$') as shard_index,
+        REGEXP_CONTAINS(tables.table_name, '^.+[0-9]{8}$') and tables.table_type = 'BASE TABLE' as is_date_shard,
+        case tables.table_type
+            when 'BASE TABLE' then 'table'
+            when 'VIEW' then 'view'
+            when 'MATERIALIZED VIEW' then 'materialized view'
             else 'external'
         end as table_type,
-        tables.type = 1 as is_table,
+        tables.table_type = 'BASE TABLE' as is_table,
         JSON_VALUE(table_description.option_value) as table_comment,
-        tables.size_bytes,
-        tables.row_count
-    from {{ information_schema.replace(information_schema_view='__TABLES__') }} tables
-    left join {{ information_schema.replace(information_schema_view='MATERIALIZED_VIEWS') }} materialized_views
-        on materialized_views.table_catalog = tables.project_id
-        and materialized_views.table_schema = tables.dataset_id
-        and materialized_views.table_name = tables.table_id
+        partitions.total_logical_bytes as size_bytes,
```

Comment:
@dbeatty10 @amychen1776 It looks like you're saying that we're not getting the size nor the row count. Are these two fields (`partitions.total_logical_bytes` and `partitions.total_rows`) coming through as 0?

Reply:
I haven't tried out this PR, so I'm not sure how `partitions.total_logical_bytes` and `partitions.total_rows` are coming through. How are they coming through for you @mikealfare ?

It would be awesome if they successfully preserve non-null and non-zero values for `size_bytes` and `total_rows`!

Reply:
Sounds like we need an integration test or two. I know I worked on this when I thought this was the source of a bug a few weeks ago. It turned out to not be the case though, so I left it there. @jtcohen6 @amychen1776, would you consider prioritizing this for the upcoming sprint so that we could capture the value here? We would get two main things:
- get off of `__TABLES__`, which is an undocumented dependency (we should be using `TABLES`
- reduce permissions required to run `dbt-bigquery`